packages/wallet/src/redux/protocols/direct-funding/state.ts

```diff
@@ -1,111 +1,111 @@
   safeToDepositLevel: string;
-  depositStatus?: depositing.DepositStatus;
-  channelFundingStatus: ChannelFundingStatus;
+  type: ChannelFundingStatus;
   requestedTotalFunds: string;
   requestedYourContribution: string;
   channelId: string;
   ourIndex: number;
 }
 export interface NotSafeToDeposit extends BaseDirectFundingState {
-  channelFundingStatus: typeof NOT_SAFE_TO_DEPOSIT;
+  type: typeof NOT_SAFE_TO_DEPOSIT;
+}
+export interface WaitForDepositTransaction extends BaseDirectFundingState {
+  type: typeof WAIT_FOR_DEPOSIT_TRANSACTION;
+  transactionSubmissionState: TransactionSubmissionState;
 }
-export interface WaitForFundingConfirmation extends BaseDirectFundingState {
-  depositStatus: typeof depositing.DEPOSIT_CONFIRMED;
-  channelFundingStatus: typeof SAFE_TO_DEPOSIT;
+export interface WaitForFundingAndPostFundSetup extends BaseDirectFundingState {
+  type: typeof WAIT_FOR_FUNDING_AND_POST_FUND_SETUP;
+  channelFunded: boolean;
+  postFundSetupReceived: boolean;
 }
-export interface ChannelFunded extends BaseDirectFundingState {
-  depositStatus: depositing.DepositStatus;
-  channelFundingStatus: typeof CHANNEL_FUNDED;
+export interface FundingSuccess extends BaseDirectFundingState {
+  type: typeof FUNDING_SUCCESS;
 }
 // constructors
-export function baseDirectFundingState<T extends BaseDirectFundingState>(
-  params: T,
+export function baseDirectFundingState(
+  params: Properties<BaseDirectFundingState>,
 ): BaseDirectFundingState {
   const {
     requestedTotalFunds,
     requestedYourContribution,
     channelId,
     ourIndex,
     safeToDepositLevel,
-    channelFundingStatus,
+    type: channelFundingStatus,
   } = params;
   return {
     requestedTotalFunds,
     requestedYourContribution,
     channelId,
     ourIndex,
     safeToDepositLevel,
-    channelFundingStatus,
+    type: channelFundingStatus,
+  };
+}
+export function notSafeToDeposit(params: Properties<BaseDirectFundingState>): NotSafeToDeposit {
+  return {
+    ...baseDirectFundingState(params),
+    type: NOT_SAFE_TO_DEPOSIT,
   };
 }
-export function notSafeToDeposit<T extends BaseDirectFundingState>(params: T): NotSafeToDeposit {
+export function waitForDepositTransaction(
+  params: Properties<BaseDirectFundingState>,
+  transactionSubmissionState: TransactionSubmissionState,
+): WaitForDepositTransaction {
   return {
     ...baseDirectFundingState(params),
-    channelFundingStatus: NOT_SAFE_TO_DEPOSIT,
+    type: WAIT_FOR_DEPOSIT_TRANSACTION,
+    transactionSubmissionState,
   };
 }
-export function waitForFundingConfirmed<T extends BaseDirectFundingState>(
-  params: T,
-): WaitForFundingConfirmation {
+
+interface ConditionalParams {
+  channelFunded: boolean;
+  postFundSetupReceived: boolean;
+}
+export function waitForFundingAndPostFundSetup<T extends BaseDirectFundingState>(
+  params: Properties<BaseDirectFundingState>,
+  conditionalParams: ConditionalParams,
+): WaitForFundingAndPostFundSetup {
```

Comment:
A possible simplification here is to use `Properties` to pull in every property apart from `type` from the `WaitForFundingAndPostFundSetup`:
```ts
export function waitForFundingAndPostFundSetup(
    params: Properties<WaitForFundingAndPostFundSetup>
): WaitForFundingAndPostFundSetup {
    const { channelFunded, postFundSetupReceived } = params;
    return {
        ...baseDirectFundingState(params),
        type: WAIT_FOR_FUNDING_AND_POST_FUND_SETUP, 
        channelFunded,
        postFundSetupReceived,
    };
}
```
There's also now [a utility](https://github.com/magmo/apps/blob/master/packages/wallet/src/redux/utils.ts#L8-L16) to help with this pattern, allowing you to write it:
```ts
const waitForFundingAndPostFundSetup: Constructor<WaitForFundingAndPostFundSetup> = params => {
     // ....
}
```

Reply:
Thanks for pointing this out. Addressed in https://github.com/magmo/apps/commit/3e7331214023bb2201c6aa913bfa5b45e09c1c27.